napari_aicssegmentation/widgets/collapsible_box.py

```diff
@@ -0,0 +1,98 @@
+from pathlib import Path
+
+from qtpy.QtGui import QPixmap
+from qtpy.QtWidgets import (
+    QFrame,
+    QHBoxLayout,
+    QLabel,
+    QVBoxLayout,
+    QWidget
+)
+
+DIR = Path.cwd() / "napari_aicssegmentation"
+
+
+"""
+A collapsible box widget containing a title box and a content box. The title box can be clicked
+to toggle the visibility of the content box.
+
+Params:
+    step:       Number representing the workflow step. This will be displayed as part of the
+                    title and be used as a part of the object name for identification.
+    title:      String
+    content:    QLayout to be nested inside the content box
+    isOpen:     Boolean, whether the widget is open or collapsed
+    isEnabled:  Boolean, overwrites the built-in property QWidget.enabled property
+"""
+
+
+class CollapsibleBox(QWidget):
+    def __init__(self, step, title, content, isOpen=True, isEnabled=True):
+        super().__init__()
+        self.step = step
+        self.title = title
+        self.content = content
+        self.isOpen = isOpen
+        self.setEnabled(isEnabled)
+
+        self.layout = QVBoxLayout()
+        self.layout.setContentsMargins(0, 11, 0, 11)
+        self.layout.setSpacing(0)   # No space between title_box and content_box
+        self.setLayout(self.layout)
+        # This will probably come in handy when we're trying to manage multiple CollapsibleBox objects
+        self.setObjectName(f"collapsibleBox{step}")
```

Comment:
Same as above - I think we don't need the concept of a `step` in here, so this becomes obsolete.